kvmagent/kvmagent/plugins/nvram/nvram.py

```
@@ -0,0 +1,86 @@
+            extension.prepare_nvram_before_vm_start()
+        else:
+            extension = nvram_local.LocalNvRamVmExtensions()
+            extension.nvram_volume = self.nvram_volume
+            extension.vm_uuid = self.vm_uuid
+            extension.nvram_install_path = self.nvram_install_path
+            extension.prepare_nvram_before_vm_start()
+    
+    def cleanup(self):
+        if self.vm_uuid == '':
+            if self.nvram_install_path == '':
+                raise Exception('vm_uuid and nvram_install_path in NvRamVmExtensions cannot both be empty.')
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)